test/unit/spec/LoginRouter_spec.js

```diff
@@ -823,65 +823,63 @@ Expect.describe('LoginRouter', function() {
         expect(form.isRecoveryQuestion()).toBe(true);
       });
   });
-  itp('navigates to PrimaryAuth and shows a flash error if the stateToken expires', function() {
-    return setup({}, resRecovery)
-      .then(function(test) {
-        Util.mockRouterNavigate(test.router);
-        test.setNextResponse(resRecovery);
-        test.router.refreshAuthState('dummy-token');
-        return Expect.waitForRecoveryQuestion(test);
-      })
-      .then(function(test) {
-        test.setNextResponse(errorInvalidToken);
-        const form = new RecoveryForm($sandbox);
 
-        form.setAnswer('4444');
-        form.submit();
-        return Expect.waitForPrimaryAuth(test);
-      })
-      .then(function(test) {
-        expect(test.afterErrorHandler).toHaveBeenCalledTimes(1);
-        expect(test.afterErrorHandler.calls.allArgs()).toEqual([
-          [
-            { controller: 'primary-auth' },
-            {
-              name: 'AuthApiError',
-              message: 'Invalid token provided',
-              statusCode: 401,
-              xhr: {
-                status: 401,
-                responseType: 'json',
-                responseText: '{"errorCode":"E0000011","errorSummary":"Invalid token provided","errorLink":"E0000011","errorId":"oaeuiUWCPr6TUSkOclgVGlWqw","errorCauses":[]}',
-                responseJSON: {
-                  errorCode: 'E0000011',
-                  errorSummary: 'Invalid token provided',
-                  errorLink: 'E0000011',
-                  errorId: 'oaeuiUWCPr6TUSkOclgVGlWqw',
-                  errorCauses: [],
-                },
-              },
+  itp('navigates to PrimaryAuth and shows a flash error if the stateToken expires', async function() {
+    // flashError is set in RouterUtil, shown in courage BaseForm method: __showErrors
+    // BaseLoginRouter will see the flashError and trigger the error on PrimaryAuth model after render 
+    const test = await setup({}, resRecovery);
+    Util.mockRouterNavigate(test.router);
+    test.setNextResponse(resRecovery);
+    test.router.refreshAuthState('dummy-token');
+    await Expect.waitForRecoveryQuestion(test);
+    
+    test.setNextResponse(errorInvalidToken);
+    let form = new RecoveryForm($sandbox);
+
+    form.setAnswer('4444');
+    form.submit();
+    await Expect.waitForPrimaryAuth(test);
```

Comment:
Since we aren't handling the resolved/rejected values for these anymore, should we wrap these tests in a `try catch` or ensure all expectations are called and there are no silent failures?

Reply:
if any exception throws within the test it will fail